tests/main/snap-icons-fallback/task.yaml

```
@@ -5,6 +5,40 @@ details: |
     then the icons API should fall back to serving the icon from the snap's
     store listing, which snapd should have cached during the snap install.
 
+environment:
+    # in the with-icon-fallback variant the icons are downloaded as expected
+    CLOUD/with_icon_fallback: ""
+
+    # in the no-icon-fallback variant no icons are downloaded, due to executing
+    # within a 'cloud' environment
+    CLOUD/no_icon_fallback: "cloudlet"
+
+prepare: |
+    systemctl stop snapd.socket
+    gojq .data.config.core.cloud /var/lib/snapd/state.json > cloud.backup
+
+    if [ "$SPREAD_VARIANT" = "with_icon_fallback" ]; then
+        gojq .data.config.core.cloud={} /var/lib/snapd/state.json > state.json."$SPREAD_VARIANT"
```

Might not be necessary, but bash might think `{}` are special shell group options, e.g. `/foo/{bar,baz}`

```suggestion
        gojq '.data.config.core.cloud={}' /var/lib/snapd/state.json > state.json."$SPREAD_VARIANT"
```